src/Confluent.Kafka/Impl/Metadata.cs

```diff
@@ -90,4 +90,61 @@ struct rd_kafka_group_list
         internal IntPtr groups;
         internal int group_cnt;
     };
+    
+    enum rd_kafka_vtype
+    {
+        End,       // va-arg sentinel
+        Topic,     // (const char *) Topic name
+        Rkt,       // (rd_kafka_topic_t *) Topic handle
+        Partition, // (int32_t) Partition
+        Value,     // (void *, size_t) Message value (payload)
+        Key,       // (void *, size_t) Message key
+        Opaque,    // (void *) Application opaque
+        MsgFlags,  // (int) RD_KAFKA_MSG_F_.. flags
+        Timestamp, // (int64_t) Milliseconds since epoch UTC
+        Header,    // (const char *, const void *, ssize_t) Message Header
+        Headers,   // (rd_kafka_headers_t *) Headers list
+    }
+
+    [StructLayout(LayoutKind.Sequential)]
+    struct rd_kafka_key_val
```

Comment:
this name implies to me there is a correspondingly named struct in librdkafka, which there isn't. maybe call it `ptr_and_size`?